src/components/WeaponComponent.vue

```diff
@@ -198,6 +227,85 @@ export default {
 		.camouflage {
 			user-select: none;
 
+			&.weapon-layout-grid > .inner {
+				flex-direction: column;
+				justify-content: center;
+
+				&.completed > .info {
+					opacity: 0.5;
+				}
+
+				img {
+					height: 100%;
+					object-fit: cover;
+					position: relative;
+					width: 100%;
+					z-index: 1;
+				}
+
+				.icon-component {
+					left: 50%;
+					opacity: 0;
+					position: absolute;
+					transform: translate(-50%, -50%);
+					transition: $transition;
+					top: 50%;
+					z-index: 2;
+				}
+
+				.info {
+					padding: 8px;
+
+					.name {
+						font-size: 14px;
+					}
+				}
+			}
+
+			&.weapon-layout-list > .inner {
+				$image-size: 30px;
```

Comment:
We can slightly decrease the camouflage-image size
```suggestion
				$image-size: 25px;
```